internal/transactions/sign.go

```diff
@@ -54,22 +56,43 @@ func sign(
 	services *services.Services,
 	state *flowkit.State,
 ) (command.Result, error) {
-	filename := args[0]
-	payload, err := readerWriter.ReadFile(filename)
+	var payload []byte
+	var err error
+	yesOption := globalFlags.Yes
+	filenameOrUrl := args[0]
+
+	if signFlags.FromRemoteUrl {
```

Comment:
I guess it's too un-reliable just to try and figure this out from the input itself? Probably safer with a flag, with only slightly worse UX. Probably worth the trade off, but wanted to mention it's probably possible.

Reply:
I agree with above 100%, but I would add that we should avoid such usage of flags. I think it would be better suited to use the flag like: `flow transactions sign --from-remote-url <url>` instead of using it to indicate what the argument value is, usually flags are meant to either indicate a boolean value (with them being present or not) or to pass the value but doing so directly using the flag (as in my example), I think if you want to indicate the argument value is remote URL then the flag should be more like `--is-remove-url` but not sure if that is better, I would say it's better to just use the flag to provide the value and then make the argument value optional if that flag is present.